public/templates/templates.go

```diff
@@ -99,9 +116,173 @@ func (t *Templater) RenderTemplate(w io.Writer, data interface{}, mainTmpl Templ
 		"len": func(a string) int {
 			return len(a)
 		},
+		"lenA": func(a []string) int {
+			return len(a)
+		},
+		"memberAddPermission": func(id int) bool {
+			u, err := t.User.GetUser(context.Background(), user.User{UserID: id})
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+
+			p, err := t.User.GetPermissionsForUser(context.Background(), u)
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+
+			for _, perm := range p {
+				if perm == permissions.ManageMembersMembersAdd.GetString() {
+					return true
+				}
+			}
+			return false
+		},
+		"checkPermission": func(id int, p string) bool {
+			m := GetValidPermissions(permissions.Permissions(p))
+
+			u, err := t.User.GetUser(context.Background(), user.User{UserID: id})
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+
+			p1, err := t.User.GetPermissionsForUser(context.Background(), u)
+			if err != nil {
+				fmt.Println(err)
+				return false
+			}
+
+			for _, perm := range p1 {
+				if m[perm] {
+					return true
+				}
+			}
+			return false
+		},
 	})
 
-	t1, err = t1.ParseFS(tmpls, "_base.tmpl", "_head.tmpl", "_footer.tmpl", "_navbar.tmpl", "_sidebar.tmpl", string(mainTmpl))
+	t1, err = t1.ParseFS(tmpls, "_base.tmpl", "_head.tmpl", "_footer.tmpl", "_navbar.tmpl", "_sidebar.tmpl", mainTmpl.GetString())
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	return t1.Execute(w, data)
+}
+
+func (t *Templater) RenderTemplatePagination(w io.Writer, data interface{}, mainTmpl Template) error {
```

Comment:
This seems like it duplicates a lot of `RenderTemplate` - was there a reason they can't be the same function?

Reply:
I think I did it like that as it was nicer in code and was easier to read, the function map was later moved into just one function so it was shortened down

Reply:
Thanks, I'll resolve this for now but will come back to it in a later PR.